routers/web/repo/issue_suggestions.go

```diff
@@ -0,0 +1,78 @@
+package repo
+
+import (
+	"net/http"
+
+	"code.gitea.io/gitea/models/db"
+	issues_model "code.gitea.io/gitea/models/issues"
+	issue_indexer "code.gitea.io/gitea/modules/indexer/issues"
+	"code.gitea.io/gitea/services/context"
+)
+
+type issueSuggestion struct {
+	ID          int64  `json:"id"`
+	Title       string `json:"title"`
+	State       string `json:"state"`
+	PullRequest *struct {
+		Merged bool `json:"merged"`
+		Draft  bool `json:"draft"`
+	} `json:"pull_request,omitempty"`
+}
+
+// IssueSuggestions returns a list of issue suggestions
+func IssueSuggestions(ctx *context.Context) {
+	keyword := ctx.Req.FormValue("q")
+
+	searchOpt := &issue_indexer.SearchOptions{
+		Paginator: &db.ListOptions{
+			Page:     0,
+			PageSize: 5,
+		},
+		Keyword:  keyword,
+		RepoIDs:  []int64{ctx.Repo.Repository.ID},
```

Comment:
I think we can support both `#123` and `<owner>/<repo>#123` at the same PR.

Reply:
And we should also support both `#<keyword>` and `#<digitals>`, if it's `<digitals>`, parse as int64 and get the issue directly. And it should be the first one on the returned results.

Reply:
- `<owner>/<repo>#123` would require the user to enter the correct repo name already, thereby I assume most users would simply copy the complete to the issue of that other repo.
- For `#<digitals` I don't think this is helpful as you would need to know the issue id already to search for it, so you could directly write it, right 😅  

As both options could be easily added later on, I would skip adding them for now.

Reply:
> I think we can support both `#123` and `<owner>/<repo>#123` at the same PR. Looks like Github doesn't support `<owner>/<repo>#123` yet.

I do not think it makes sense, what is the real use case to find some issues by crossing repositories?

(I also agree adding `#<digitals>` to the top is handful while it is better to have but not a must in this PR)